### PR TITLE
Fix RLBot's virtual environments

### PR DESCRIPTION
**Pull request by VirxEC**
Changes:

- clear by default is False
- Set with_pip to true (auto-install pip and setuptools for us)
- Use `python -m pip -U pip` to upgrade pip
- Make the create method configure using system site-packages before calling `post_setup`

### CODE REVIEW

File: src/main/python/rlbot/utils/virtual_environment_management.py [hunk 1]
```diff
@@ -28,8 +59,9 @@ def post_setup(self, context: SimpleNamespace) -> None:
         sys.stderr.write(f'Installing {requirements}...\n')
         sys.stderr.flush()
 
-        args = [context.env_exe, '-m', 'ensurepip']
-        finished_process = self.run_and_dump(args, timeout=120)
+        # Actually update pip
+        args = [context.env_exe, '-m', 'pip', 'install', '-U', 'pip']
+        finished_process = self.run_and_dump(args, timeout=300)
```
Comment (tarehart) [inline src/main/python/rlbot/utils/virtual_environment_management.py:64]: What's the story on this? Upgrading pip seems like a good idea, but is it safe to remove `env_exe -m ensurepip`? Are we guaranteed to always already have pip at this point?

File: src/main/python/rlbot/utils/virtual_environment_management.py [hunk 2]
```diff
@@ -12,10 +12,41 @@
...
+        Create a virtual environment in a directory.
+        Changed so system site packages is configured before post_setup is ran.
+
+        :param env_dir: The target directory to create an environment in.
+        """
+
+        env_dir = os.path.abspath(env_dir)
+        context = self.ensure_directories(env_dir)
+        # See issue 24875. We need system_site_packages to be False
+        # until after pip is installed.
```
Comment (tarehart) [inline src/main/python/rlbot/utils/virtual_environment_management.py:33]: Interesting! Looks like the referenced issue is https://bugs.python.org/issue24875. It's old and allegedly resolved, do we still need this code?

File: src/main/python/rlbot/utils/virtual_environment_management.py (hunk 1 above)
Comment (VirxEC) [reply src/main/python/rlbot/utils/virtual_environment_management.py:64]: Yes, with_pip guarantees that we will.

File: src/main/python/rlbot/utils/virtual_environment_management.py (hunk 2 above)
Comment (VirxEC) [reply src/main/python/rlbot/utils/virtual_environment_management.py:33]: That comment is actually from the original code in Python, I just copy+pasted the function and made post_setup() get called last. https://github.com/python/cpython/blob/3.7/Lib/venv/__init__.py#L52-L76

**Review by tarehart — APPROVED** [review]
Looks good!

**Pull request merged**